datafusion/core/src/datasource/file_format/mod.rs

```diff
@@ -235,7 +239,7 @@ pub(crate) enum AbortMode {
 }
 
 /// A wrapper struct with abort method and writer
-struct AbortableWrite<W: AsyncWrite + Unpin + Send> {
+pub(crate) struct AbortableWrite<W: AsyncWrite + Unpin + Send> {
```

Comment:
this module is getting pretty large -- I think it would be easier to navigate it if were split into some smaller modules (e.g. put the write related code into `datafusion/core/src/datasource/file_format/write.rs`, for example). Any interest in doing so @devinjdangelo  or @metesynnada ? If not, I can give it a try

Reply:
That is a good idea. The code is likely to become more complex / long with additional helpers as we get into parallelization and dealing with parquet. I'm working on a PR to add support for inserting parquet now, and I can consolidate the shared write code into a new module as part of that.

Reply:
Opened a draft here #7244